src/main/java/org/alfresco/repo/domain/permissions/ADMAccessControlListDAO.java

```diff
@@ -459,52 +459,62 @@ else if (dbAcl.getAclType() == ACLType.SHARED)
-        if (!asyncCall)
-        {
-            // make regular method call
-            setFixedAcls(nodeId, inheritFrom, mergeFrom, sharedAclToReplace, changes, set, asyncCall, propagateOnChildren);
-            return true;
-        }
-        else
-        {
-            // check transaction time
-            long transactionStartTime = AlfrescoTransactionSupport.getTransactionStartTime();
-            long transactionTime = System.currentTimeMillis() - transactionStartTime;
+    
+    /**
+     * Adds ASPECT_PENDING_FIX_ACL aspect to nodes when transactionTime reaches max admitted time
+     * MNT-18308: No longer checks if call is async in order to evaluate time passed to decide if nodes should be
+     * processed by job. This is now the default behavior for both sync and async calls: when fixedAclMaxTransactionTime
+     * is exceeded, call turns async and all remaining nodes should be processed by job FixedACLUpdater
+     * 
+     * @param nodeId
+     *            the nodeId of the current node being processed
+     * @param inheritFrom
+     *            the parent node's ACL
+     * @param mergeFrom
+     *            the new shared ACL, if already known.
+     * @param sharedAclToReplace
+     *            the old shared ACL, to be replaced
+     * @param changes
+     *            the list in which to record changes
+     * @param set
+     *            if to set the shared ACL on the parent
+     * @param asyncCall
+     *            if the call was initially set as async
+     * @param propagateOnChildren
+     *            current setting of child propagation of ACL creation
+     * @return new setting on child propagation of ACL creation
+     * 
+     */
+    private boolean setFixAclPending(Long nodeId, Long inheritFrom, Long mergeFrom, Long sharedAclToReplace,
+            List<AclChange> changes, boolean set, boolean asyncCall, boolean propagateOnChildren)
+    {
+        // check transaction time
+        long transactionStartTime = AlfrescoTransactionSupport.getTransactionStartTime();
+        long transactionTime = System.currentTimeMillis() - transactionStartTime;
+        if (transactionTime < fixedAclMaxTransactionTime)
+        {
+            // make regular method call if time is under max transaction configured time
+            setFixedAcls(nodeId, inheritFrom, mergeFrom, sharedAclToReplace, changes, set, asyncCall, propagateOnChildren);
+            return true;
+        }
+
+        // If flag is still unset or false, the call until now has been sync and will turn async, we should throw a
+        // warning
+        if (log.isWarnEnabled() && (AlfrescoTransactionSupport.getResource(FixedAclUpdater.FIXED_ACL_ASYNC_REQUIRED_KEY) == null
+                || (Boolean) AlfrescoTransactionSupport.getResource(FixedAclUpdater.FIXED_ACL_ASYNC_REQUIRED_KEY) == false))
+        {
+            log.warn("Time was exceeded in transaction " + AlfrescoTransactionSupport.getTransactionId()
```

Comment:
I would like to propose to change the message to be a bit more precise:
"The ACL processing time in transaction X exceeded the configured limit of Y ms. The rest of the ACL processing will be done asynchronously."
I guess it is ok to omit some of the information, as the elapsed time (transactionTime) will be very close to the limit and the exact date when this happened will be in the log statement anyways.